### PR TITLE
Update to Go 1.22.

### PR DESCRIPTION
**Pull request by shaunrd0**
This updates to use Go 1.22, since Go 1.21 [reached EOL 4 months ago](https://endoflife.date/go).

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
Now that `min` is available, can you use it in these cases?

https://github.com/TileDB-Inc/TileDB-Go/blob/f20011a937ba1ca3acef7d934e947c6339adb2c4/buffer.go#L138-L143

https://github.com/TileDB-Inc/TileDB-Go/blob/f20011a937ba1ca3acef7d934e947c6339adb2c4/buffer.go#L174-L180

**Review by teo-tsirpanis — APPROVED** [review]
Thanks!

**Review by NullHypothesis — APPROVED** [review]

**Comment by shaunrd0** [comment]
@TileDB-Inc/tiledb-cloud-core Blocked on merging this because of required CI checks for Go 1.20. Can someone with access check on the required workflows for TileDB-Go to replace 1.20 with 1.22? I think there is likely a branch protection rule or something in repository settings that I don't have access to

**Comment by NullHypothesis** [comment]
> @TileDB-Inc/tiledb-cloud-core Blocked on merging this because of required CI checks for Go 1.20. Can someone with access check on the required workflows for TileDB-Go to replace 1.20 with 1.22? I think there is likely a branch protection rule or something in repository settings that I don't have access to

I remember running into this problem in the past. Can you try opening a new PR?

**Comment by teo-tsirpanis** [comment]
I don't think it needs a new PR. Someone with permissions (@Shelnutt2?) needs to update the branch protection rules.

**Comment by shaunrd0** [comment]
> I remember running into this problem in the past. Can you try opening a new PR?

Yep no luck.. worth a shot though! https://github.com/TileDB-Inc/TileDB-Go/pull/362

I'll just reopen this and close the new PR. Sorry for all the noise :sweat_smile:

**Comment by NullHypothesis** [comment]
> I don't think it needs a new PR. Someone with permissions (@Shelnutt2?) needs to update the branch protection rules.

@snagles is also known to have superpowers.

**Comment by Shelnutt2** [comment]
@shaunrd0 and @NullHypothesis you have leveled up to "maintainer". New quests have been unlocked in the "settings arena". Go forth and tame the CI settings.

**Comment by teo-tsirpanis** [comment]
It [needs](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#:~:text=Manage%20branch%20protection%20rules%20and%20repository%20rulesets) admin permissions.

**Comment by shaunrd0** [comment]
> It [needs](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#:~:text=Manage%20branch%20protection%20rules%20and%20repository%20rulesets) admin permissions.

:+1: Yep, now I can access the Settings panel but can't view or modify branch protection rules.

**Pull request merged**
